### PR TITLE
ADD: wrapper for locator

### PR DESCRIPTION
**Pull request by Stalkcomrade**

### CODE REVIEW

**Comment by andy-thomason** [comment]
Thanks for making a contribution. Looks good to me, @yutannihilation can comment.

**Comment by yutannihilation** [comment]
Thanks! Could you also tweak `haveLocator`?

**Comment by Stalkcomrade** [comment]
Sure!

**Comment by Stalkcomrade** [comment]
The only comment -- I am not sure what the default should be for USE_LOCATOR.
I set it to `true` as it looks similar to capture.

I saw that `haveCaptrue` depends on the internals of the device being used
https://github.com/wch/r-source/blob/f09d3d7fa4af446ad59a375d914a0daf3ffc4372/src/library/grDevices/src/devWindows.c#L3423
https://github.com/wch/r-source/blob/9f284035b7e503aebe4a804579e9e80a541311bb/src/modules/X11/devX11.c#L2959

**Comment by yutannihilation** [comment]
> I am not sure what the default should be for USE_LOCATOR.
> I set it to `true` as it looks similar to capture.

Thanks, I too am not yet sure what value is suitable for the default, but `true` should be good for now. As long as we provide an empty implementation, it won't crash the user's R session at least.

File: extendr-api/src/graphics/device_driver.rs [hunk 1]
```diff
@@ -265,6 +270,15 @@ pub trait DeviceDriver: std::marker::Sized {
         0
     }
 
+    /// A callback function that returns the coords of the event
+    fn locator(&mut self, x: &mut f64, &mut y: f64, dd: DevDesc) {}
+
+    /// A callback function for X11_eventHelper.
+    /// Argument `code` should, ideally, be of type c_int,
+    /// but compiler throws erors. It should be ok to use
+    /// i32 here.
```
Comment (yutannihilation) [inline extendr-api/src/graphics/device_driver.rs:279]: Curious.

**Review by yutannihilation — COMMENTED** [review]
Looks good in general, but I'd like to investigate the problem of `c_int` and `i32`.

Minor requests:

* Could you remove `.DS_Store`?
* Could you apply `cargo fmt`?

File: extendr-api/src/graphics/device_driver.rs (hunk 1 above)
Comment (Stalkcomrade) [reply extendr-api/src/graphics/device_driver.rs:279]: Sorry for keeping this unfixed for that long.
I will try to get back to this in the upcoming week -- had some issues with running github actions locally via act.

Comment (yutannihilation) [reply extendr-api/src/graphics/device_driver.rs:279]: No worries! Please feel free to ask for any help.

Comment (Stalkcomrade) [reply extendr-api/src/graphics/device_driver.rs:279]: Done.
The commit history looks a bit awkward though.

Comment (Stalkcomrade) [reply extendr-api/src/graphics/device_driver.rs:279]: I have the following error, which I cannot resolve.
I'm not even sure why `u32` is expected in the first place.

*Output*:
> Error[E0308]: mismatched types
>    --> /home/stlk/projects/oss/extendr/extendr-api/src/graphics/device_driver.rs:794:42
>     |
> 794 |             (*p_dev_desc).locator = Some(device_driver_locator::<T>);
>     |                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u32`, found `()`
>     |
>     = note: expected fn pointer `unsafe extern "C" fn(_, _, _) -> u32`
>                   found fn item `unsafe extern "C" fn(_, _, _) {device_driver_locator::<T>}`
> 
> For more information about this error, try `rustc --explain E0308`.

**Comment by yutannihilation** [comment]
Sorry for the delay! https://github.com/extendr/extendr/pull/389/commits/8f34cad1571029c39b97a532171e6ac4861c8ec7 should fix the error. You needed to implement `locator()` to return `bool` to match the signature.

**Comment by yutannihilation** [comment]
And, sorry, it seems I broke the diffs...🙄

**Comment by andy-thomason** [comment]
@yutannihilation Feel free to merge this if you think it is good.

**Comment by yutannihilation** [comment]
@Stalkcomrade 
May I merge this? There's some room for improvement (especially in comments), but I think it's okay to merge before conflicting with other changes.

**Comment by Stalkcomrade** [comment]
@yutannihilation, sure, thank you!

**Comment by yutannihilation** [comment]
Then, I'm merging this. Thanks so much for your contribution!

**Pull request merged**